tracer/build/_build/Build.Steps.cs

```diff
@@ -1368,156 +1289,182 @@ _ when path.Contains("dependency-libs") => false,
             DotnetBuild(projects, framework: Framework);
         });
 
-    Target CompileSamplesWindows => _ => _
+    Target CompileSamples => _ => _
+        .Description("Compiles all the sample projects")
         .Unlisted()
+        .After(Clean)
         .DependsOn(HackForMissingMsBuildLocation)
-        .After(CompileDependencyLibs)
-        .After(CompileFrameworkReproductions)
-        .Requires(() => MonitoringHomeDirectory != null)
-        .Requires(() => Framework)
         .Executes(() =>
         {
             if (TestAllPackageVersions)
             {
-                // TODO this is hacky as I couldn't figure out what was going on here so I opted to just delete everything each time
-                //      for some reason projects that declare specific TargetFrameworks in the project file
-                //      will duplicate their package versions
-                //      e.g. Samples.GraphQL4\bin\4.1.0\Debug\net7.0\bin\4.3.0
-                //      this will go on and create a ton of folders/files
-                //      my hacky workaround for this at the moment is to simply remove the bin/obj directories beforehand
-                //      GrpcDotNet, GraphQL4, and HotChocolate samples had this issue
-                Logger.Information("Cleaning up sample projects that use multiple package versions");
-
-                var multiPackageProjects = new List<string>();
-                var samplesFile = BuildDirectory / "PackageVersionsGeneratorDefinitions.json";
-                using var fs = File.OpenRead(samplesFile);
-                var json = JsonDocument.Parse(fs);
-                multiPackageProjects = json.RootElement
-                                           .EnumerateArray()
-                                           .Select(e => e.GetProperty("SampleProjectName").GetString())
-                                           .Distinct()
-                                           .Where(name => name switch
-                                           {
-                                               "Samples.MySql" => false, // the "non package version" is _ALSO_ tested separately
-                                               _ => true
-                                           })
-                                           .ToList();
-                var patterns = new List<string>();
-
-                foreach (var dir in multiPackageProjects)
-                {
-                    patterns.Add($"test/test-applications/integrations/{dir}/bin");
-                    patterns.Add($"test/test-applications/integrations/{dir}/obj");
-                }
+                // Explicitly compiles the prerequisites which MSbuild doesn't
+                // (no, I don't know why, I can't seem to convince it to do this automatically)
```

Comment:
not really important, but did you figure out what caused the projects to duplicate themselves when running with `TestAllPackageVersions` it doesn't seem to happen anymore with this so that is 👍

Reply:
No, I didn't see it happening so tbh, I didn't _try_ to figure it out 😅